internal/ingressclass/ingressclass.go

```diff
@@ -54,6 +56,11 @@ func matches(objIngressClass, contourIngressClass string) bool {
 		return objIngressClass == "" || objIngressClass == DefaultClassName
 	}
 
-	// Otherwise, the object's ingress class must match Contour's.
-	return objIngressClass == contourIngressClass
+	// Otherwise, the object's ingress class must match one of Contour's.
+	for _, contourIngressClassEntry := range strings.Split(contourIngressClass, ",") {
```

Comment:
let's add a test case to https://github.com/projectcontour/contour/blob/main/internal/ingressclass/ingressclass_test.go#L26 and https://github.com/projectcontour/contour/blob/main/internal/ingressclass/ingressclass_test.go#L111 for a comma-separated `contourIngressClass`.